### PR TITLE
Initial checkin

### PR DESCRIPTION
**Pull request by jacobbaungard**
This is the first check-in of code to the Merlin Container Poller.

Part of: MON-12873

Signed-off-by: Jacob Hansen <jhansen@op5.com>

### CODE REVIEW

File: images/daemon/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ENV NAEMON_CFG=/etc/naemon/ MERLIN_CFG=/etc/merlin/ \
+    MERLIN_SOCK=/var/lib/merlin/ipc.sock \
+    NAEMON_UID=1001 NAEMON_GID=0
+
+COPY shared_files/install.sh /usr/local/bin/install.sh
+COPY daemon/default_config /usr/local/etc/merlin/default_config
+RUN bash /usr/local/bin/install.sh
+# Get rid of default merlin.conf, we'll replace by the default_config
+# during the entry script
+#RUN rm $MERLIN_CFG/merlin.conf
```
Comment (kdnilsson) [inline images/daemon/Dockerfile]: Maybe remove if it really is a comment?

File: deployment_examples/k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      terminationGracePeriodSeconds: 120
+      containers:
+       - name: naemon-core
+         image: op5com/merlin-naemon:latest
+         imagePullPolicy: Never
+         resources:
+         volumeMounts:
+          - mountPath: /var/lib/merlin/
+            name: ipc
+          - mountPath: /opt/monitor/op5/merlin
```
Comment (kdnilsson) [inline deployment_examples/k8s.yaml]: op5? . Is the mountPath correct?

File: images/shared_files/install.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+useradd -l -g "$NAEMON_GID" -u "$NAEMON_UID" -s /bin/bash -M naemon -d /var/lib/naemon/
+
+curl https://download.opensuse.org/repositories/home:/itrs-op5/CentOS_7/home:itrs-op5.repo -o /etc/yum.repos.d/merlin.repo
+curl https://download.opensuse.org/repositories/home:/naemon/CentOS_7/home:naemon.repo -o /etc/yum.repos.d/naemon.repo
+yum install -y naemon-core naemon-livestatus merlin-slim monitor-merlin-slim merlin-apps-slim && \
+  yum clean all && \
+  rm -rf /var/cache/yum /etc/yum.repos.d/op5-release.repo
```
Comment (kdnilsson) [inline images/shared_files/install.sh]: Do we ever install the op5-release package?. If not I think the /etc/yum.repos.d/op5-release.repo could be removed

**Review by kdnilsson — COMMENTED** [review]
Some comments. Otherwise I think it looks good. Have not tested to run anything

File: images/daemon/Dockerfile (hunk 1 above)
Comment (jacobbaungard) [reply images/daemon/Dockerfile]: Thanks, done. It was moved to install.sh.

**Comment by jacobbaungard** [comment]
Thanks! Pushed correction to those.

**Review by kdnilsson — APPROVED** [review]
Looks good. Nice job!

**Pull request merged**